### PR TITLE
[READY]BM SPEEDUWAGON, PUSH IT TO THE LIMIT

### PR DESCRIPTION
**Pull request by Militaires**
:cl: BASILMAN YOUR MAIN MAN
add: Added BM SPEEDWAGON THE BEST (AND ONLY) SPACE CAR ON THE MARKET.
/:cl:

# I PRESENT THE BM SPEEDWAGON

THE FRONTIER IN VEHICLE TECHNOLOGY, DRIVE ON THE PURE POWER OF THE STARS, RUN PEOPLE OVER, MOVE REALLY FAST DOWN THE HALL, PICK UP HOES*

dreammaker_2017-01-24_20-35-06

windscreen removed for clarity
dreammaker_2017-01-24_20-35-17

this is intended for wizards at some point but ill leave that for another pr because i fear the balance meme flocks.

this was fun to make, and hopefully fun to watch operate.

[]: # i think this should be added because it's fun to run people over and to drive on space, plus, we need this to be better than vg who have the firebird pontiac

eliminates #23170

### CODE REVIEW

**Comment by Arzion5** [comment]
https://s-media-cache-ak0.pinimg.com/564x/45/f8/ae/45f8ae084978e881ef501e316df2afb4.jpg

**Comment by Militaires** [comment]
@arzion5

wizard version with skull and firetrails confirmed to be coming soon.

**Comment by Iamgoofball** [comment]
i should bring back the clown car

File: code/modules/vehicles/speedbike.dm [hunk 1]
```diff
@@ -34,4 +34,29 @@
...
+		A.throw_at(throw_target, 4, 3)
+		A.Weaken(5)
+		A.adjustStaminaLoss(30)
+		A.apply_damage(rand(20,35), BRUTE)
+		visible_message("<span class='danger'>[src] crashes into [A]!</span>")
+		playsound(src, 'sound/effects/bang.ogg', 50, 1)
+
+/obj/vehicle/space/speedbike/speedwagon/buckle_mob(mob/living/M, force = 0, check_loc = 1)
+ 	. = ..()
+		riding_datum = new/datum/riding/space/speedbike/speedwagon
```
Comment (Iamgoofball) [inline code/modules/vehicles/speedbike.dm]: # no drifting no buy

**Comment by Kromgar** [comment]
I remember playing /vg/ once and was xenoarcheologist and then i just foind the wizard car. I drove it around the station and it could phase through walls you doin that with wiz car?

**Comment by Militaires** [comment]
@Kromgar the wiz version will be 12 kinds of crazy

File: code/modules/vehicles/speedbike.dm (hunk 1 above)
Comment (Militaires) [inline code/modules/vehicles/speedbike.dm]: while i get that you want drifting how is the piece of code you reviewed any relevant

**Comment by Tacolizard** [comment]
this is dumb because you can just get my Atvmemes from cargo

**Comment by Militaires** [comment]
@Tacolizard but your atvmemes

- suck dick

- cant run people over

- slower than my grandma

- cant drive on space

- not sweet-looking it's like a golf cart without the top.

**Comment by Bill-Luxe** [comment]
I just put it here...
https://www.youtube.com/watch?v=cyqwrO09iq4

**Comment by Supermichael777** [comment]
pics with driver

**Comment by Tacolizard** [comment]
@Militaires What the fuck did you just fucking say about me, you little shit? Ill have you know I graduated top of my class in the Navy Seals, and Ive been involved in numerous secret raids on Al-Quaeda, and I have over 300 confirmed kills. I am trained in gorilla warfare and Im the top sniper in the entire US armed forces. You are nothing to me but just another target. I will wipe you the fuck out with precision the likes of which has never been seen before on this Earth, mark my fucking words. You think you can get away with saying that shit to me over the Internet? Think again, fucker. As we speak I am contacting my secret network of spies across the USA and your IP is being traced right now so you better prepare for the storm, maggot. The storm that wipes out the pathetic little thing you call your life. Youre fucking dead, kid. I can be anywhere, anytime, and I can kill you in over seven hundred ways, and thats just with my bare hands. Not only am I extensively trained in unarmed combat, but I have access to the entire arsenal of the United States Marine Corps and I will use it to its full extent to wipe your miserable ass off the face of the continent, you little shit. If only you could have known what unholy retribution your little clever comment was about to bring down upon you, maybe you would have held your fucking tongue. But you couldnt, you didnt, and now youre paying the price, you goddamn idiot. I will shit fury all over you and you will drown in it. Youre fucking dead, kiddo.

**Comment by Durkel** [comment]
Clown car when?

**Comment by Supermichael777** [comment]
The wizard one needs to be jet black and on fire. Animated fire

**Comment by Iamgoofball** [comment]
the plates better be NOFATCHX

**Comment by Iamgoofball** [comment]
also red racing stripes for speed boost when?

File: code/modules/vehicles/speedbike.dm (hunk 1 above)
Comment (kevinz000) [inline code/modules/vehicles/speedbike.dm]: I could make my thing datumized and be more vareditable if you want.

**Comment by kevinz000** [comment]
@Tacolizard fuck off
@Militaires if you want drifting i can make my flightpack stuff more modular.

**Comment by Mindustry** [comment]
Someone hasn't driven an ATV on lavaland

**Comment by Militaires** [comment]
@kevinz000 go ahead but dont trust ill use or understand that shit.

@Mindustry does it have some special collision or physics on lavaland cause if so hey wow look at me everybody that doesnt exist on this lava hellhole i will run you over.

**Comment by jughu** [comment]
Better have a honking command

**Comment by Supermichael777** [comment]
S P E E D W A G O N

**Comment by SgtxRootbeer** [comment]
Can I shove a meta-friend/hostage into the passenger seat?

**Comment by Militaires** [comment]
@Supermichael777 
shh dont let them

@SgtxRootbeer 
if someone figures out how to do multiple buckle vehicles then yeah but until then im fuked cause how man it's borky

**Comment by kevinz000** [comment]
@Militaires Oh yeah your riding datums are done, I'm gonna use em to make riding cyborgs a thing.

**Comment by PKPenguin321** [comment]
I like how so far nobody has questioned the very idea of adding a car to a space atmospherics simulator

**Comment by Jordie0608** [comment]
Compared to ratvar, narsie, lavaland, the singularity and devils; a car is very mundane.

**Comment by KingPhilipIII** [comment]
I better have the right of way when a car and I come to an intersection.

**Comment by GunHog** [comment]
You have the right to become space-roadkill.

**Comment by KingPhilipIII** [comment]
Does this function the same way vehicle buckles on secways and the Janicart work or no? Because hijacking someone's car via unbuckle-shove-steal keys would be annoying as hell.

**Comment by KorPhaeron** [comment]
If riding vechicles flat out doesn't work again after this (fourth?) PR about it I'm just going to start closing your (obviously untested) PRs fair warning

**Comment by Militaires** [comment]
@KorPhaeron  
>flat out
>flat out doesn't work
>Again

where the fuck are you getting your info from? 

riding vehicles ALWAYS worked since i datumized it, riding MOBS was the thing that failed and it runtimed like hell,, there's multiple prs cause i fixed mob riding twice, closed a pr to merge it with this one which makes 4 in total, none of those prs fixed vehicles because vehicles always worked even if they runtimed.

where the fuck are you getting your info from.

you dont make multiple prs to fix runtimes, decide to close some of them and have one fail because of conflicts and there's some maintainer who bugs you to fix the runtimes (rightfully so cause i caused them) so you go and do it. then the next cocksucker tells you that your datum flat out never worked with info from god's know's ass.

if you weren't kor i'd tell you to try closing my (obviously untested) dick because im having a bad day man.

**Comment by Militaires** [comment]
fuck it im just crabby sorry about that.

**Comment by Militaires** [comment]
i forgot to add

**Comment by KorPhaeron** [comment]
Nice

**Comment by Tacolizard** [comment]
@kevinz000 

Hows it goin?
The names, Foster... 💀
Im a Demon, Sent from the wastlands of the Glowing Sea, where we were cast out, and through disaster and devostation, we lost our innocense. It is in that moment, that feeling, the feeling of powerlessness. It is in that feeling where good men, turn cruel. I on the other hand, know this feeling, and will do everything in my power, to make sure no man must bere that feeling, alone.
So step up, if you want to get burn't 🔥
"This cosmic dance﻿ of bursting decadence and withheld permissions twists all our arms collectively, but if sweetness can win, and it can, then I'll still be here tomorrow, to high five you yesterday my friend. Peace" - The Tart Toater
Now Fuck off
...................../´¯/)
....................,/¯../
.................../..../
............./´¯/'...'/´¯¯`·¸
........../'/.../..../......./¨¯\
........('(...´...´.... ¯~/'...')
..........................'...../
..........''............. _.·´
..........................(
..............................

**Comment by lzimann** [comment]
Make a separated PR to fix that issue(that broke all the vehicles) instead of including in a feature PR.

**Comment by Militaires** [comment]
@lzimann done please remov tag

File: code/modules/vehicles/speedbike.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/vehicles/speedbike.dm]: drifting code is shit so I wouldnt' force it onto anyone until kevinz00 fixes his shit

Comment (Iamgoofball) [inline code/modules/vehicles/speedbike.dm]: not that kind of drifting

**Pull request merged**